sources/github.py

```
@@ -193,6 +199,79 @@ def _fetch_my_repo_issues():
+                "url": subject_url or repo_info.get("html_url", ""),
+                "type": "notification",
+                "notification_reason": reason,
+                "notification_id": notif.get("id", ""),
+                "latest_comment_url": latest_comment_url,
+                "createdAt": notif.get("updated_at", ""),
+            })
+        elif subject_type == "CheckSuite":
+            # CI failure with no PR link. Build a synthetic number from
+            # the title so it de-duplicates against itself but never collides
+            # with a real PR/issue number.
+            hash_val = int(hashlib.md5(title.encode()).hexdigest(), 16) % (10**6)
```

Ruff (0.16.2)

[error] Undefined name `hashlib`

(F821)